src/api/firebase.js

```diff
@@ -80,11 +82,38 @@ export async function addItem(listId, { itemName, daysUntilNextPurchase }) {
 
 export async function updateItem(listId, itemId) {
 	const docRef = doc(db, listId, itemId);
+	// We need to use some data from the document for our future purchase estimates.
+	const docSnap = await getDoc(docRef);
+	const today = new Date();
 
-	await updateDoc(docRef, {
-		dateLastPurchased: new Date(),
-		totalPurchases: increment(1),
-	});
+	if (docSnap.exists()) {
+		const data = docSnap.data();
```

Comment:
So there's an unnecessary call to firebase here just to get the item attributes again. How can this be written better? Do we really need a check to docSnap.exists if the list is showing our list item?